libs/pika/async_mpi/include/pika/async_mpi/trigger_mpi.hpp

```diff
@@ -186,6 +177,46 @@ namespace pika::mpi::experimental::detail {
                                     r.op_state.status, PIKA_MOVE(r.op_state.receiver));
                                 break;
                             }
+                            case handler_method::new_task:
+                            {
+                                // The callback will call set_value/set_error inside a new task
+                                // and execution will continue on that thread
+                                add_new_task_request_callback(
+                                    r.op_state.request, p, PIKA_MOVE(r.op_state.receiver));
+                                break;
+                            }
+                            case handler_method::continuation:
+                            {
+                                // The callback will call set_value/set_error
+                                // execution will continue on the callback thread
+                                add_continuation_request_callback<>(r.op_state.request, r.op_state);
+                                break;
+                            }
+                            case handler_method::mpix_continuation:
```

Comment:
What happens if MPI continuations are not enabled, but one requests MPI continuations?

How much of the MPI continuations stuff can we ifdef out if they're not enabled/available?

Reply:
at the moment, it's an error. the code will break, but that's why I put the `#pragma warning` in the other place. it's experimental and I didn't try to make it safe yet.